src/resolver/environment.cc

```diff
@@ -129,6 +129,25 @@ Environment::groupIsLocked( const InstallDescriptors & group,
 
   for ( auto & [iid, descriptor] : group )
     {
+      /* Check for upgrades. */
+      if ( bool * upgradeEverything = std::get_if<bool>( &this->upgrades ) )
+        {
+          /* If we're upgrading everything, the group needs to be locked again.
+           */
+          if ( *upgradeEverything ) { return false; }
+        }
+      else
+        {
+          /* If the current iid is being upgraded, the group needs to be locked
+           * again.
+           */
```

Comment:
```suggestion
          /* If the current iid is being upgraded, the group needs to be
           * locked again. */
```

I'm just nick picking, don't mind me <3